src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -259,6 +252,49 @@ trait ChoosesMode {
             Vector()
         }
 
+      val chosenCurrentTourMode: Option[BeamTourMode] = personData.currentTourMode match {
+        case Some(tourMode) => Some(tourMode)
+        case None =>
+          val availablePersonalVehicleModes = availablePersonalStreetVehicles.map(x => x.streetVehicle.mode).distinct
+          val availableFirstAndLastLegModes =
+            availablePersonalVehicleModes.flatMap(x => BeamTourMode.enabledModes.get(x)).flatten
+          val modesToQuery =
+            (availablePersonalVehicleModes ++ BeamMode.nonPersonalVehicleModes ++ availableFirstAndLastLegModes).distinct
+              .intersect(availableModes)
+          val dummyVehicleType = beamScenario.vehicleTypes(dummyRHVehicle.vehicleTypeId)
+          val currentTour = _experiencedBeamPlan.getTourContaining(nextAct)
+          val tourModeCosts = beamServices.skims.od_skimmer.getTourModeCosts(
+            modesToQuery,
+            currentTour,
+            dummyRHVehicle.vehicleTypeId,
+            dummyVehicleType,
+            beamScenario.fuelTypePrices(dummyVehicleType.primaryFuelType)
+          )
+          val modeToTourMode =
```

Comment:
@Xuan-1998 So as I understand it you're suggesting that the problem here is that if the agents have access to a shared bike when they hit this logic, then they end up potentially choosing a `BIKE_BASED` tour, even though they need to leave the bike behind after the first trip? That definitely seems like it could be causing the problem. Perhaps we need to add some other logic that only allows a `BIKE_BASED` tour if they have access to a personal bike (and similar for personal vs shared cars)

Reply:
And just thinking through a to do here -- we don't currently factor in walking distance to the available street vehicles into these costs. Perhaps eventually we should call the skimmer to factor in those access/egress costs as well. Don't think it's worth it for now, though.